### PR TITLE
Add Devise

### PR DESCRIPTION
**Pull request by Kanyorok**
- Removed `current_user` method in `ApplicationController` as devise provided one.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
  - Can register a new user.
  - User logs in with a combination of email and password.
  - Hashed password should be stored in the database.
  - Ask for confirmation of email.
  - Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and my styling.

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  <div class="field">
+    <%= f.label :photo %><br />
+    <%= f.text_field :photo %>
+  </div>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autocomplete: "email" %>
+  </div>
```
Comment (Munsa1) [inline app/views/devise/registrations/new.html.erb]: - [ ] KIndly ensure signing up is possible with the app, currently this is not happening. Please look at the screencast below for more details.
[cinnamon-2023-12-14T130148+0200.webm](https://github.com/Kanyorok/PiquePosts/assets/81932478/0d92aa2d-fcda-4902-9045-0c89466fc076)

Further check another screencast showing that the app is not working as expected.
[cinnamon-2023-12-14T132401+0200.webm](https://github.com/Kanyorok/PiquePosts/assets/81932478/267b6fbb-e920-416a-b75a-200982e52429)

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            <% end %>
+          </div>
+      </div>
+      <div class="actions">
+        <%= f.submit "Log in", type: "button", class: "btn btn-primary btn-block mb-4"%>
+      </div>
+    <% end %>
+  </section> 
+</div>
+<%= render "devise/shared/links" %>
```
Comment (Munsa1) [inline app/views/devise/sessions/new.html.erb]: - [ ] Kindly ensure your `Log in` button spans the same width as the input fields so as to match the given design

Given frame 
image

Your design
image

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (Munsa1) [inline app/views/devise/registrations/new.html.erb]: - [ ] Kindly ensure the sign up page follows the given wireframe for `Login` page as stated in the requirements. i.e style it like you have styled the login page with the contents of the page centred

Your design
image

Given frame
image

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Kanyorok, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Set Up Devise :heavy_check_mark: 
Correct Git Flow :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by danmainah — APPROVED** [review]
Hi @Kanyorok and @Al-Amin-Khan-Shakil , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 

## Highlights
✔️  Implemented the change requested to ensure the users are able to login to the app.
✔️  Implemented the change requested to ensure the registration and login pages styling match the wireframe design given.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**